Slim/Web/Time.pm

```diff
@@ -63,17 +84,19 @@ sub tzAPIsuccess {
 	my $res = eval { from_json($http->content) };
 	if ($@ || ref $res ne 'HASH') {
 		$log->error($@ || 'Invalid JSON response: ' . $http->content);
-		$res = {}; # guarantee that $res will be a hash ref
+		_sendHTTPresponse($httpClient, $response, RC_INTERNAL_SERVER_ERROR, '');
+		return;
 	}
 
 	# '$tz' will hold the Olson formatted TimeZone to be returned.
 	# Setting '$tz' to '' signals a validation failure, and triggers a
-	# 404 error response.
+	# 204 (No content) response.
 
 	my $tz = $res->{'timezone'};
 	if (!defined $tz || ref $tz) {
```

Comment:
You should remove the `defined` and check for any falsy value: if the returned result was an empty string, we can stop right here.

Reply:
Yes, now done.